Consider enforcing non-Rust dependencies via a build script

stratisd requires cryptsetup version 2.3.0. We expect that downstream packagers will enforce this via packaging. However, when building from source in the presence of cryptsetup version 2.2.0, the result is a compiler error which is shown below. Although we document the mandatory dependency, we should also consider a build script which enforces it, to avoid any possible confusion among users building from source. But, if we enforce this dependency via a build script it makes sense to also enforce our libblkid dependency since running with too low a version of libblkid would be expected to result in stratisd failing to locate existing Stratis devices on startup.

```
 Compiling libstratis v2.0.1 (/home/mulhern/stratis2/stratisd)
error[E0599]: no function or associated item named `alloc` found for struct `libcryptsetup_rs::mem::SafeMemHandle` in the current scope
   --> src/engine/strat_engine/backstore/crypt.rs:693:45
    |
693 |         let mut key_buffer = SafeMemHandle::alloc(STRATIS_KEY_SIZE)?;
    |                                             ^^^^^ function or associated item not found in `libcryptsetup_rs::mem::SafeMemHandle`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
error: could not compile `libstratis`.

To learn more, run the command again with --verbose.
make: *** [Makefile:50: build] Error 101
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,7 +15,10 @@ language: rust
 # Use a package which supplies cryptsetup 2.3.0 which is required
 # by features added in libcryptsetup-rs that are used in stratisd.
 before_install:
+  - sudo add-apt-repository -y "deb http://us.archive.ubuntu.com/ubuntu/ eoan main" 
   - sudo apt-get -q update
+  # Update to a more recent version of blkid
+  - sudo apt-get -y install util-linux libblkid-dev
   # cryptsetup-bin conflicts with custom built cryptsetup
   - sudo apt-get remove cryptsetup-bin
   - sudo apt-get install -y libargon2-0 libjson-c3

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -3,6 +3,7 @@ name = "libstratis"
 version = "2.0.1"
 authors = ["Stratis Developers <stratis-devel@lists.fedorahosted.com>"]
 edition = "2018"
+build = "build.rs"
 
 [dependencies]
 dbus = {version = "0.8", optional = true}
@@ -30,6 +31,9 @@ libcryptsetup-rs = "0.3.0"
 [dependencies.uuid]
 version = "0.7"
 features = ["serde", "v4"]
+
+[build-dependencies]
+pkg-config = "0.3"
 
 [dev-dependencies]
 error-chain = "0.12.1"

--- a/build.rs
+++ b/build.rs
@@ -1,0 +1,17 @@
+use pkg_config::Config;
+
+fn main() {
+    if let Err(e) = Config::new().atleast_version("2.3.0").find("libcryptsetup") {
+        panic!(
+            "At least version 2.3.0 of cryptsetup is required to compile stratisd: {}",
+            e
+        );
+    }
+
+    if let Err(e) = Config::new().atleast_version("2.32.0").find("blkid") {
+        panic!(
+            "At least version 2.32.0 of blkid is required to compile stratisd: {}",
+            e
+        );
+    }
+}

